Index: BAL/WKAL/Concretizations/Widgets/Gtk/BCFrameGtk.cpp
===================================================================
--- BAL/WKAL/Concretizations/Widgets/Gtk/BCFrameGtk.cpp (revision 474)
+++ BAL/WKAL/Concretizations/Widgets/Gtk/BCFrameGtk.cpp (working copy)
@@ -23,7 +23,8 @@
 
 #include "config.h"
 #include "Frame.h"
-
+#include "FramePrivate.h"
+#include "UserStyleSheetLoader.h"
 #include "NotImplemented.h"
 
 namespace WKAL {
@@ -34,4 +35,21 @@
     return 0;
 }
 
+void Frame::setUserStyleSheetLocation(const KURL& url)
+{
+    delete d->m_userStyleSheetLoader;
+    d->m_userStyleSheetLoader = 0;
+    if (d->m_doc && d->m_doc->docLoader())
+        d->m_userStyleSheetLoader = new UserStyleSheetLoader(d->m_doc, url.string());
+}
+
+void Frame::setUserStyleSheet(const String& styleSheet)
+{
+    delete d->m_userStyleSheetLoader;
+    d->m_userStyleSheetLoader = 0;
+    if (d->m_doc)
+        d->m_doc->setUserStyleSheet(styleSheet);
+}
+
+
 }
