Index: BAL/OWBAL/Concretizations/Types/WK/BCKURLWK.cpp
===================================================================
--- BAL/OWBAL/Concretizations/Types/WK/BCKURLWK.cpp	(revision 306)
+++ BAL/OWBAL/Concretizations/Types/WK/BCKURLWK.cpp	(working copy)
@@ -30,7 +30,7 @@
 #include "PlatformString.h"
 #include "TextEncoding.h"
 
-#if USE(ICU_UNICODE)
+#if USE(ICU_UNICODE) && !USE(BALI18N)
 #include <unicode/uidna.h>
 #elif USE(QT4_UNICODE)
 #include <QUrl>
