Index: BAL/OWBAL/Concretizations/ImageDecoder/GIF/WK/BCGIFImageDecoderWK.cpp
===================================================================
--- BAL/OWBAL/Concretizations/ImageDecoder/GIF/WK/BCGIFImageDecoderWK.cpp	(revision 2800)
+++ BAL/OWBAL/Concretizations/ImageDecoder/GIF/WK/BCGIFImageDecoderWK.cpp	(working copy)
@@ -27,7 +27,7 @@
 #include "GIFImageDecoder.h"
 #include "GIFImageReader.h"
 
-#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX)
+#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX) || PLATFORM(BAL)
 
 namespace OWBAL {
 
