Index: BAL/OWBAL/Concretizations/Types/WK/BCThreadSpecificWK.h
===================================================================
--- BAL/OWBAL/Concretizations/Types/WK/BCThreadSpecificWK.h  (revision 395)
+++ BAL/OWBAL/Concretizations/Types/WK/BCThreadSpecificWK.h  (working copy)
@@ -29,6 +29,7 @@
 #ifndef WTF_ThreadSpecific_h
 #define WTF_ThreadSpecific_h

+#include <wtf/Assertions.h>
 #include <wtf/Noncopyable.h>

 #if USE(PTHREADS) || PLATFORM(WIN)
