Index: BAL/OWBAL/Concretizations/Types/Gtk/BCThreadingGtk.cpp
===================================================================
--- BAL/OWBAL/Concretizations/Types/Gtk/BCThreadingGtk.cpp	(revision 486)
+++ BAL/OWBAL/Concretizations/Types/Gtk/BCThreadingGtk.cpp	(working copy)
@@ -40,7 +40,7 @@
 
 Mutex* atomicallyInitializedStaticMutex;
 
-#if !PLATFORM(DARWIN)
+#if !PLATFORM(DARWIN) || PLATFORM(MACPORT)
 static ThreadIdentifier mainThreadIdentifier;
 #endif
 
@@ -60,7 +60,7 @@
         atomicallyInitializedStaticMutex = new Mutex;
         threadMapMutex();
         wtf_random_init();
-#if !PLATFORM(DARWIN)
+#if !PLATFORM(DARWIN) || PLATFORM(MACPORT)
         mainThreadIdentifier = currentThread();
 #endif
 #if PLATFORM(WIN)
