Index: BAL/OWBAL/Concretizations/Facilities/Posix/BCFileSystemPosix.cpp
===================================================================
--- BAL/OWBAL/Concretizations/Facilities/Posix/BCFileSystemPosix.cpp	(revision 2800)
+++ BAL/OWBAL/Concretizations/Facilities/Posix/BCFileSystemPosix.cpp	(working copy)
@@ -157,4 +157,34 @@
     return dirname(fsRep.mutableData());
 }
 
+//FIXME : implement this and send a patch to WebCore
+int writeToFile(PlatformFileHandle, const char* data, int length)
+{
+    return -1;
+}
+
+CString fileSystemRepresentation(const String &file)
+{
+    return file.utf8();
+}
+
+void closeFile(PlatformFileHandle&)
+{
+}
+
+bool unloadModule(PlatformModule)
+{
+    return false;
+}
+
+String homeDirectoryPath()
+{
+    return getenv("HOME");
+}
+
+CString openTemporaryFile(const char* prefix, PlatformFileHandle&)
+{
+    return "";
+}
+
 } // namespace WebCore
