Index: BAL/WKAL/Concretizations/Fonts/WK/BCFontWK.cpp
===================================================================
--- BAL/WKAL/Concretizations/Fonts/WK/BCFontWK.cpp	(revision 2823)
+++ BAL/WKAL/Concretizations/Fonts/WK/BCFontWK.cpp	(working copy)
@@ -250,7 +250,7 @@
 {
     if (currentCharacter + 1 < m_end) {
         if (combiningClass(m_run[currentCharacter + 1]) == hiraganaKatakanaVoicingMarksCombiningClass) {
-#if USE(ICU_UNICODE)
+#if USE(ICU_UNICODE) && !USE(BALI18N)
             // Normalize into composed form using 3.2 rules.
             UChar normalizedCharacters[2] = { 0, 0 };
             UErrorCode uStatus = U_ZERO_ERROR;  
@@ -263,6 +263,7 @@
             QString res = tmp.normalized(QString::NormalizationForm_C, QChar::Unicode_3_2);
             if (res.length() == 1)
                 return res.at(0).unicode();
+#elif USE(ICU_UNICODE) && USE(BALI18N)
 #endif
         }
     }
