Index: BAL/WKAL/Concretizations/Graphics/Cairo/BCImageCairo.cpp
===================================================================
--- BAL/WKAL/Concretizations/Graphics/Cairo/BCImageCairo.cpp	(révision 498)
+++ BAL/WKAL/Concretizations/Graphics/Cairo/BCImageCairo.cpp	(copie de travail)
@@ -36,6 +36,24 @@
 #include <cairo.h>
 #include <math.h>
 
+#if PLATFORM(AMIGAOS4)
+#include "owb-config.h"
+
+Vector<char> loadResourceIntoArray(const char* name)
+{
+    WebCore::String fullPath = RESOURCE_PATH;
+    WebCore::String extension = ".png";
+    fullPath += name;
+    fullPath += extension;
+    
+    static PassRefPtr<WebCore::SharedBuffer> buffer = WebCore::SharedBuffer::createWithContentsOfFile(fullPath);
+    Vector<char> array;
+    if (buffer)
+        array = buffer.get()->buffer();
+    return array;
+}
+#endif
+
 namespace WKAL {
 
 void FrameData::clear()
@@ -180,6 +198,25 @@
     m_isSolidColor = false;
 }
 
+#if PLATFORM(AMIGAOS4)
+void BitmapImage::initPlatformData()
+{
 }
 
+void BitmapImage::invalidatePlatformData()
+{
+}
+
+PassRefPtr<Image> Image::loadPlatformResource(const char *name)
+{
+    Vector<char> arr = loadResourceIntoArray(name);
+    RefPtr<BitmapImage> img = BitmapImage::create();
+    RefPtr<SharedBuffer> buffer = SharedBuffer::create(arr.data(), arr.size());
+    img->setData(buffer, true);
+    return img.release();
+}
+#endif
+
+}
+
 #endif // PLATFORM(CAIRO)
