Index: BAL/WKAL/Concretizations/Widgets/Gtk/PasteboardHelperGtk.cpp
===================================================================
--- BAL/WKAL/Concretizations/Widgets/Gtk/PasteboardHelperGtk.cpp        (revision 474)
+++ BAL/WKAL/Concretizations/Widgets/Gtk/PasteboardHelperGtk.cpp        (working copy)
@@ -20,31 +20,33 @@
 #include "PasteboardHelperGtk.h"
 
 #include "Frame.h"
-#include "webkitwebframe.h"
+/*#include "webkitwebframe.h"
 #include "webkitwebview.h"
-#include "webkitprivate.h"
+#include "webkitprivate.h"*/
 
 #include <gtk/gtk.h>
 
 using namespace WebCore;
 
-namespace WebKit
-{
+namespace WebKit {
 
 GtkClipboard* PasteboardHelperGtk::getClipboard(Frame* frame) const {
-    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
+/*    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
     return gtk_widget_get_clipboard(GTK_WIDGET (webView),
-                                    GDK_SELECTION_CLIPBOARD);
+                                    GDK_SELECTION_CLIPBOARD);*/
+    return 0;
 }
 
 GtkTargetList* PasteboardHelperGtk::getCopyTargetList(Frame* frame) const {
-    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
-    return webkit_web_view_get_copy_target_list(webView);
+/*    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
+    return webkit_web_view_get_copy_target_list(webView);*/
+    return 0;
 }
 
 GtkTargetList* PasteboardHelperGtk::getPasteTargetList(Frame* frame) const {
-    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
-    return webkit_web_view_get_paste_target_list(webView);
+/*    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
+    return webkit_web_view_get_paste_target_list(webView);*/
+    return 0;
 }
 
 }
