Index: BAL/OWBAL/Concretizations/ImageDecoder/PNG/WK/BCPNGImageDecoderWK.cpp
===================================================================
--- BAL/OWBAL/Concretizations/ImageDecoder/PNG/WK/BCPNGImageDecoderWK.cpp	(revision 2800)
+++ BAL/OWBAL/Concretizations/ImageDecoder/PNG/WK/BCPNGImageDecoderWK.cpp	(working copy)
@@ -40,7 +40,7 @@
 #include "png.h"
 #include "assert.h"
 
-#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX)
+#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX) || PLATFORM(BAL)

 #if COMPILER(MSVC)
 // Remove warnings from warning level 4.
