Index: BAL/WKAL/Concretizations/Network/Curl/BCResourceHandleManagerCurl.cpp
===================================================================
--- BAL/WKAL/Concretizations/Network/Curl/BCResourceHandleManagerCurl.cpp	(revision 286)
+++ BAL/WKAL/Concretizations/Network/Curl/BCResourceHandleManagerCurl.cpp	(working copy)
@@ -467,8 +467,12 @@
         outData = reinterpret_cast<char*>(g_base64_decode(data.utf8().data(), &outLength));
 #else
         Vector<char> out;
-        if (base64Decode(data.latin1().data(), data.length(), out))
-            data = String(out.data(), out.size());
+        if (base64Decode(data.latin1().data(), data.length(), out)) {
+            outData = (char*)malloc(out.size());
+            for (size_t i = 0; i < out.size(); i++)
+                outData[i] = out[i];
+            outLength = out.size();
+        }
         else
             data = String();
 #endif
@@ -498,6 +502,9 @@
 
 #ifdef USE_GLIB_BASE64
     g_free(outData);
+#else
+    if(outData)
+        free(outData);
 #endif
 
     client->didFinishLoading(handle);
