Index: BAL/WKAL/Concretizations/Network/Curl/BCResourceHandleManagerCurl.cpp
===================================================================
--- BAL/WKAL/Concretizations/Network/Curl/BCResourceHandleManagerCurl.cpp	(revision 515)
+++ BAL/WKAL/Concretizations/Network/Curl/BCResourceHandleManagerCurl.cpp	(working copy)
@@ -507,8 +507,11 @@
 
     if (outData)
         client->didReceiveData(handle, outData, outLength, 0);
-    else
-        client->didReceiveData(handle, out.data(), out.size(), 0);
+    else 
+        if (out.size() > 0)
+            client->didReceiveData(handle, out.data(), out.size(), 0);
+        else
+            client->didReceiveData(handle, data.latin1().data(), data.length(), 0);
 
 #ifdef USE_GLIB_BASE64
     g_free(outData);
