Index: BAL/WKAL/Concretizations/Widgets/WK/BCScrollViewWK.cpp
===================================================================
--- BAL/WKAL/Concretizations/Widgets/WK/BCScrollViewWK.cpp	(revision 580)
+++ BAL/WKAL/Concretizations/Widgets/WK/BCScrollViewWK.cpp	(working copy)
@@ -77,6 +77,10 @@
 
 void ScrollView::setHasHorizontalScrollbar(bool hasBar)
 {
+#if PLATFORM(AMIGAOS4)
+    if (!parent())
+        hasBar = false;
+#endif
     if (hasBar && !m_horizontalScrollbar && !platformHasHorizontalAdjustment()) {
         m_horizontalScrollbar = Scrollbar::createNativeScrollbar(this, HorizontalScrollbar, RegularScrollbar);
         addChild(m_horizontalScrollbar.get());
@@ -88,6 +92,10 @@
 
 void ScrollView::setHasVerticalScrollbar(bool hasBar)
 {
+#if PLATFORM(AMIGAOS4)
+    if (!parent())
+        hasBar = false;
+#endif
     if (hasBar && !m_verticalScrollbar && !platformHasVerticalAdjustment()) {
         m_verticalScrollbar = Scrollbar::createNativeScrollbar(this, VerticalScrollbar, RegularScrollbar);
         addChild(m_verticalScrollbar.get());
