Index: BAL/OWBAL/Concretizations/Types/WK/BCTCSpinLockWK.h
===================================================================
--- BAL/OWBAL/Concretizations/Types/WK/BCTCSpinLockWK.h	(révision 427)
+++ BAL/OWBAL/Concretizations/Types/WK/BCTCSpinLockWK.h	(copie de travail)
@@ -37,7 +37,11 @@
 
 #include <time.h>       /* For nanosleep() */
 
+#if PLATFORM(AMIGAOS4)
+#include <unistd.h>
+#else
 #include <sched.h>      /* For sched_yield() */
+#endif
 
 #if HAVE(STDINT_H)
 #include <stdint.h>
@@ -135,7 +139,9 @@
 #define SPINLOCK_INITIALIZER { 0 }
 
 static void TCMalloc_SlowLock(volatile unsigned int* lockword) {
+#if !PLATFORM(AMIGAOS4)
   sched_yield();        // Yield immediately since fast path failed
+#endif
   while (true) {
     int r;
 #if COMPILER(GCC)
@@ -181,6 +187,8 @@
     // Sleep for a few milliseconds
 #if PLATFORM(WIN_OS)
     Sleep(2);
+#elif PLATFORM(AMIGAOS4)
+    usleep(200);
 #else
     struct timespec tm;
     tm.tv_sec = 0;
