Index: BAL/OWBAL/Concretizations/Types/Gtk/BCThreadingGtk.cpp
===================================================================
--- BAL/OWBAL/Concretizations/Types/Gtk/BCThreadingGtk.cpp      (revision 474)
+++ BAL/OWBAL/Concretizations/Types/Gtk/BCThreadingGtk.cpp      (working copy)
@@ -63,7 +63,9 @@
 #if !PLATFORM(DARWIN)
         mainThreadIdentifier = currentThread();
 #endif
+#if PLATFORM(WIN)
         initializeMainThread();
+#endif
     }
 }
 
