Index: BAL/OWBAL/Concretizations/ImageDecoder/BMP/WK/BCBMPImageDecoderWK.cpp
===================================================================
--- BAL/OWBAL/Concretizations/ImageDecoder/BMP/WK/BCBMPImageDecoderWK.cpp	(revision 2800)
+++ BAL/OWBAL/Concretizations/ImageDecoder/BMP/WK/BCBMPImageDecoderWK.cpp	(working copy)
@@ -26,7 +26,7 @@
 #include "config.h"
 #include "BMPImageDecoder.h"
 
-#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX)
+#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX) || PLATFORM(BAL)
 
 namespace OWBAL{
 
