Index: BAL/OWBAL/Concretizations/ImageDecoder/GIF/WK/BCGIFImageReaderWK.cpp
===================================================================
--- BAL/OWBAL/Concretizations/ImageDecoder/GIF/WK/BCGIFImageReaderWK.cpp	(revision 2800)
+++ BAL/OWBAL/Concretizations/ImageDecoder/GIF/WK/BCGIFImageReaderWK.cpp	(working copy)
@@ -78,7 +78,7 @@
 #include <string.h>
 #include "GIFImageDecoder.h"
 
-#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX)
+#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX) || PLATFORM(BAL)
 
 using WebCore::GIFImageDecoder;
 
