Index: BAL/WKAL/Concretizations/Graphics/Cairo/BCImageSourceCairo.cpp
===================================================================
--- BAL/WKAL/Concretizations/Graphics/Cairo/BCImageSourceCairo.cpp	(revision 280)
+++ BAL/WKAL/Concretizations/Graphics/Cairo/BCImageSourceCairo.cpp	(working copy)
@@ -41,6 +41,8 @@
 #include "XBMImageDecoder.h"
 #endif
 
+#include <cstdio>
+
 namespace WKAL {
 
 ImageDecoder* createDecoder(const Vector<char>& data)
