Index: BAL/OWBAL/Concretizations/ImageDecoder/JPEG/WK/BCJPEGImageDecoderWK.cpp
===================================================================
--- BAL/OWBAL/Concretizations/ImageDecoder/JPEG/WK/BCJPEGImageDecoderWK.cpp	(revision 2800)
+++ BAL/OWBAL/Concretizations/ImageDecoder/JPEG/WK/BCJPEGImageDecoderWK.cpp	(working copy)
@@ -40,7 +40,7 @@
 #include <assert.h>
 #include <stdio.h>
 
-#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX)
+#if PLATFORM(CAIRO) || PLATFORM(QT) || PLATFORM(WX) || PLATFORM(BAL)
 
 extern "C" {
 #include "jpeglib.h"
