Index: BAL/OWBAL/Concretizations/Memory/WK/BCTCSystemAllocWK.cpp
===================================================================
--- BAL/OWBAL/Concretizations/Memory/WK/BCTCSystemAllocWK.cpp	(révision 427)
+++ BAL/OWBAL/Concretizations/Memory/WK/BCTCSystemAllocWK.cpp	(copie de travail)
@@ -43,8 +43,10 @@
 #else
 #include <errno.h>
 #include <unistd.h>
+#if !PLATFORM(AMIGAOS4)
 #include <sys/mman.h>
 #endif
+#endif
 #include <fcntl.h>
 #include "Assertions.h"
 #include "TCSystemAlloc.h"
