Index: BAL/OWBAL/Concretizations/Types/Pthreads/BCThreadingPthreads.cpp
===================================================================
--- BAL/OWBAL/Concretizations/Types/Pthreads/BCThreadingPthreads.cpp    (revision 474)
+++ BAL/OWBAL/Concretizations/Types/Pthreads/BCThreadingPthreads.cpp    (working copy)
@@ -59,7 +59,9 @@
 #if !PLATFORM(DARWIN)
         mainThreadIdentifier = currentThread();
 #endif
+#if PLATFORM(WIN)
         initializeMainThread();
+#endif
     }
 }
 
