Index: BAL/OWBAL/Concretizations/Memory/WK/BCTCSystemAllocWK.cpp
===================================================================
--- BAL/OWBAL/Concretizations/Memory/WK/BCTCSystemAllocWK.cpp	(revision 673)
+++ BAL/OWBAL/Concretizations/Memory/WK/BCTCSystemAllocWK.cpp	(working copy)
@@ -55,7 +55,7 @@
 
 #if HAVE(MMAP)
 static const int cProtFlags = PROT_READ | PROT_WRITE
-#if ENABLE(ASSEMBLER) && (PLATFORM(GTK) || (PLATFORM(MAC) && PLATFORM(X86_64)) || PLATFORM(QT))
+#if ENABLE(ASSEMBLER) && (PLATFORM(GTK) || (PLATFORM(MAC) && PLATFORM(X86_64)) || PLATFORM(QT) || PLATFORM(BAL))
                               | PROT_EXEC
 #endif
                               ;
