Index: BAL/WKAL/Concretizations/Widgets/WK/BCContextMenuWK.cpp
===================================================================
--- BAL/WKAL/Concretizations/Widgets/WK/BCContextMenuWK.cpp	(revision 508)
+++ BAL/WKAL/Concretizations/Widgets/WK/BCContextMenuWK.cpp	(working copy)
@@ -427,8 +427,10 @@
     if (!page)
         return;
 
+#if ENABLE(INSPECTOR)
     if (!page->inspectorController())
         return;
+#endif
 
     ContextMenuItem InspectElementItem(ActionType, ContextMenuItemTagInspectElement, contextMenuItemTagInspectElement());
     appendItem(*separatorItem());
