Index: BAL/WKAL/Concretizations/Graphics/WK/BCGraphicsContextWK.cpp
===================================================================
--- BAL/WKAL/Concretizations/Graphics/WK/BCGraphicsContextWK.cpp	(revision 2800)
+++ BAL/WKAL/Concretizations/Graphics/WK/BCGraphicsContextWK.cpp	(working copy)
@@ -434,7 +434,7 @@
 }
 #endif
 
-#if !PLATFORM(QT) && !PLATFORM(CAIRO) && !PLATFORM(SKIA)
+#if !PLATFORM(QT) && !PLATFORM(CAIRO) && !PLATFORM(SKIA) && !PLATFORM(BAL)
 void GraphicsContext::setPlatformStrokeStyle(const StrokeStyle&)
 {
 }
