Index: BAL/WKAL/Concretizations/Widgets/WK/BCWidgetWK.h
===================================================================
--- BAL/WKAL/Concretizations/Widgets/WK/BCWidgetWK.h	(révision 503)
+++ BAL/WKAL/Concretizations/Widgets/WK/BCWidgetWK.h	(copie de travail)
@@ -29,6 +29,10 @@
 
 #include <wtf/Platform.h>
 
+#if PLATFORM(BAL)
+#include "BALBase.h"
+#endif
+
 #if PLATFORM(MAC)
 #ifdef __OBJC__
 @class NSView;
@@ -198,7 +202,7 @@
     IntRect m_frame; // Not used when a native widget exists.
     PlatformWindow m_containingWindow; // Not used when a native widget exists.
 
-#if PLATFORM(MAC) || PLATFORM(GTK)
+#if PLATFORM(MAC) || PLATFORM(GTK) || PLATFORM(BAL)
     WidgetPrivate* m_data;
 #endif
 };
