Index: CookieJarCurl.cpp
===================================================================
--- CookieJarCurl.cpp	(revision 634)
+++ CookieJarCurl.cpp	(working copy)
@@ -23,23 +23,33 @@
 
 #include <wtf/HashMap.h>
 
+#include "CookieManager.h"
+
 namespace WebCore {
 
-static HashMap<String, String> cookieJar;
-
-void setCookies(Document* /*document*/, const KURL& url, const KURL& /*policyURL*/, const String& value)
+void setCookies(Document* /*document*/, const KURL& url, const KURL& policyURL, const String& value)
 {
-    cookieJar.set(url.string(), value);
+    CookieManager* manager = CookieManager::getCookieManager();
+    if (manager)
+        manager->setCookies(url, policyURL, value);
 }
 
 String cookies(const Document* /*document*/, const KURL& url)
 {
-    return cookieJar.get(url.string());
+    CookieManager* manager = CookieManager::getCookieManager();
+    if (manager)
+        return manager->getCookie(url);
+    else
+        return String();
 }
 
 bool cookiesEnabled(const Document* /*document*/)
 {
-    return true;
+    CookieManager* manager = CookieManager::getCookieManager();
+    if (manager)
+        return true;
+    else
+        return false;
 }
 
 }
