Index: ResourceHandle.h
===================================================================
--- ResourceHandle.h	(revision 634)
+++ ResourceHandle.h	(working copy)
@@ -158,6 +158,11 @@
     ResourceHandleInternal* getInternal() { return d.get(); }
 #endif
 
+#if USE(CURL)
+    void checkAndSendCookies(KURL& url);
+    void setCookies();
+#endif
+
     // Used to work around the fact that you don't get any more NSURLConnection callbacks until you return from the one you're in.
     static bool loadsBlocked();    
     
