Index: BAL/OWBAL/Concretizations/Types/WK/BCTCSpinLockWK.h
===================================================================
--- BAL/OWBAL/Concretizations/Types/WK/BCTCSpinLockWK.h	(revision 449)
+++ BAL/OWBAL/Concretizations/Types/WK/BCTCSpinLockWK.h	(working copy)
@@ -108,7 +108,7 @@
       ("isync\n\t"
        "eieio\n\t"
        "stw %1, %0"
-#if PLATFORM(DARWIN) || PLATFORM(PPC)
+#if PLATFORM(DARWIN) || (PLATFORM(PPC) && !PLATFORM(AMIGAOS4))
        : "=o" (lockword_)
 #else
        : "=m" (lockword_) 
