--- BAL/WKAL/Concretizations/Widgets/Gtk/PasteboardHelperGtk.cpp	2008-03-09 08:24:22.000000000 +0000
+++ BAL/WKAL/Concretizations/Widgets/Gtk/PasteboardHelperGtk.cpp.new	2008-04-18 14:20:08.000000000 +0000
@@ -19,9 +19,9 @@
 #include "Frame.h"
 #include "PasteboardHelperGtk.h"
 
-#include "webkitwebframe.h"
+/*#include "webkitwebframe.h"
 #include "webkitwebview.h"
-#include "webkitprivate.h"
+#include "webkitprivate.h"*/
 
 #include <gtk/gtk.h>
 
@@ -31,19 +31,22 @@
 {
 
 GtkClipboard* PasteboardHelperGtk::getClipboard(Frame* frame) const {
-    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
+/*    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
     return gtk_widget_get_clipboard(GTK_WIDGET (webView),
-                                    GDK_SELECTION_CLIPBOARD);
+                                    GDK_SELECTION_CLIPBOARD);*/
+    return 0;
 }
 
 GtkTargetList* PasteboardHelperGtk::getCopyTargetList(Frame* frame) const {
-    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
-    return webkit_web_view_get_copy_target_list(webView);
+/*    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
+    return webkit_web_view_get_copy_target_list(webView);*/
+    return 0;
 }
 
 GtkTargetList* PasteboardHelperGtk::getPasteTargetList(Frame* frame) const {
-    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
-    return webkit_web_view_get_paste_target_list(webView);
+/*    WebKitWebView* webView = webkit_web_frame_get_web_view(kit(frame));
+    return webkit_web_view_get_paste_target_list(webView);*/
+    return 0;
 }
 
 }
