Index: BAL/OWBAL/Concretizations/Internationalization/WK/BCTextEncodingWK.cpp
===================================================================
--- BAL/OWBAL/Concretizations/Internationalization/WK/BCTextEncodingWK.cpp	(revision 2823)
+++ BAL/OWBAL/Concretizations/Internationalization/WK/BCTextEncodingWK.cpp	(working copy)
@@ -75,7 +75,7 @@
     if (!length)
         return "";
 
-#if USE(ICU_UNICODE)
+#if USE(ICU_UNICODE) && !USE(BALI18N)
     // FIXME: What's the right place to do normalization?
     // It's a little strange to do it inside the encode function.
     // Perhaps normalization should be an explicit step done before calling encode.
@@ -105,6 +105,9 @@
     QString str(reinterpret_cast<const QChar*>(characters), length);
     str = str.normalized(QString::NormalizationForm_C);
     return newTextCodec(*this)->encode(reinterpret_cast<const UChar *>(str.utf16()), str.length(), handling);
+#elif USE(ICU_UNICODE) && USE(BALI18N)
+    //FIXME
+    return newTextCodec(*this)->encode(characters, length, handling);
 #endif
 }
 
